src/tiled/commandmanager.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * commandmanagar.cpp
+ * Copyright 2017, Ketan Gupta <ketan19972010@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "commandmanager.h"
+
+#include "commanddatamodel.h"
+#include "commanddialog.h"
+
+#include <QAction>
+#include <QLatin1String>
+#include <QMenu>
+#include <QSignalMapper>
+#include <QWidget>
+
+using namespace Tiled;
+using namespace Tiled::Internal;
+
+CommandManager::CommandManager(QObject *parent, QWidget *window)
+	: QObject(parent)
+	, mMainWindow(window)
+{
+	
+}
+
+CommandManager::~CommandManager()
+{
+}
+
+void CommandManager::setMainWindowMenu(QMenu *menu)
+{
+	this->mMainWindowMenu = menu;
+}
+
+void CommandManager::populateMainWindowMenu()
+{
+	populateMenu(this->mMainWindowMenu);
+}
+
+void CommandManager::showDialog()
+{
+	CommandDialog dialog(mMainWindow);
+    dialog.exec();
+}
+
+void CommandManager::populateMenu(QMenu *menu)
+{
+	menu->clear();
+
+	// Add Edit Commands first
+	QAction *mEditCommands = new QAction(this);
+    mEditCommands->setIcon(
+            QIcon(QLatin1String(":/images/24x24/system-run.png")));
+    mEditCommands->setText(tr("Edit Commands"));
```

Comment:
Please add this action as the last item in the list. Also, its text should end with `...` since it opens a dialog. Also, you should be able to re-use this `populateMenu` function in the `CommandButton` class. I think it will cause no problems to have the F5 shortcut there as well, but please verify that it won't lead to "ambiguous shortcut" problems.